pkg/util/resource/pvc.go

```diff
@@ -100,19 +100,30 @@ func CheckForSubPath(ctx context.Context, client *k8s.K8sClient, volume *v1.Pers
 		return true, nil
 	}
 	nowSubPath := volume.Spec.PersistentVolumeSource.CSI.VolumeAttributes["subPath"]
-	sc := volume.Spec.StorageClassName
+	scName := volume.Spec.StorageClassName
 
-	if sc == "" {
+	if scName == "" {
 		return false, nil
 	}
 
+	sc, err := client.GetStorageClass(ctx, scName)
+	if err == nil {
+		// if the storage class does not set the pathPattern, subpath used unique pv name, should delete
+		if _, ok := sc.Parameters["pathPattern"]; !ok {
```

Comment:
Did [`pathPattern == ""`](https://github.com/juicedata/juicefs-csi-driver/blob/b560adbb9becee15c15247eeb887cc71950e24d4/pkg/util/resource/pvc.go#L99) did the same thing? 

If pv.spec.csi.volumeAttributes (copy from sc) has no pathPattern, do not check other pvs for subPath.

Reply:
you are right, I missed it